src/main/java/org/rumbledb/compiler/TranslationVisitor.java

```diff
@@ -1129,7 +1143,9 @@ public Node visitVarDecl(JsoniqParser.VarDeclContext ctx) {
         Expression expr = null;
         if (ctx.exprSingle() != null) {
             expr = (Expression) this.visitExprSingle(ctx.exprSingle());
-            expr = new TreatExpression(expr, seq, false, expr.getMetadata());
+            if (!seq.equals(SequenceType.mostGeneralSequenceType)) {
+                expr = new TreatExpression(expr, seq, false, expr.getMetadata());
```

Comment:
I have a feeling that instead of using a boolean for shouldThrowTreatException, we could get more readable code by using an enum. 

Otherwise I would strongly recommend adding comments to describe how TreatException and TypeException relate to one another and in which cases they should be used. However, I believe refactoring with an enum could add this clarity much more subtlety, while also avoiding comments that may get outdated quickly.

Reply:
DONE, great comment, this is all done.